d8error.py

```diff
@@ -32,29 +34,36 @@ def __init__(self, etype, value):
                     prewrittenMessge = True
             self.print = prewrittenMessge
 
-        def writeRow(file):
-            """saves errors to errorLog.csv"""
-            fieldnames = ['index', 'errorType', 'errorMSG', 'feedbackRating', 'feedbackMSG', "Time"]
-            writer = csv.DictWriter(file, fieldnames=fieldnames)
+        # this generates a semi-readable summary of the traceback, which includes some information about the python code that caused the error
+        summary = traceback.format_exc().splitlines()
+        
+        # iterate through traceback object to extract linenumber and bytecode of the first two frames
+        curr_tb = tb.tb_next # skip the first frame which is the jupyter notebook frame
+        bytecodesToLinenos = []
+        while curr_tb and len(bytecodesToLinenos) < 2:
+            bytecodesToLinenos.append((curr_tb.tb_frame.f_code.co_code, curr_tb.tb_lineno))
+            curr_tb = curr_tb.tb_next
+
+
+        mode = 'w' if not os.path.isfile("errorLog.csv") else 'a'
+        if os.path.isfile("errorLog.csv") and Announce.eindex == 1:
+            with open("errorLog.csv", 'r') as f:
+                for row in csv.reader(f):
+                    self.eindex = int(row[0])
+                self.eindex += 1
+                Announce.eindex = self.eindex + 1
+        
+        with open('errorLog.csv', mode, newline='') as f:
+            fieldnames = ['index', 'errorType', 'errorMSG', 'feedbackRating', 'feedbackMSG', 'time', 'bytecodesToLinenos', 'traceSummary']
+            writer = csv.DictWriter(f, fieldnames=fieldnames)
             writer.writerow({"index": self.eindex,
                             "errorType": self.errorname,
                             "errorMSG": str(self.value),
                             "feedbackRating": self.feedbackRating,
                             "feedbackMSG": self.feedbackMSG,
-                            "Time": str(datetime.datetime.now())})
-
-        if not os.path.isfile("errorLog.csv"):
```

Comment:
Was this code meant to be deleted?

Reply:
yes i just cleaned it up to make it compatible with my changes